### DISCUSSION TITLE
Options Picker Alignment within a Form

### DISCUSSION THREAD

**Discussion by simon-lewis** [Budibase/budibase#11284] [Help] [answered]
I have to replace all my relationship fields with Options Pickers to get around the performance issues with Budibase native relationships when using an external datasource (in this case, MS SQL Server). I have to add a Data Provider nested with the Options Picker field - fine, expected - but I cant get them to align within the existing form. I've tried with and without embedded containers; the Options Picker wont stretch to fit the available panel,  FieldGroup or Form. See the screenshot attached. Can anyone point me in the right direction to align / fix the inclusion of the added options picker field within the form ?
Thanks hugely.
![image](https://github.com/Budibase/budibase/assets/92350835/be4a8e4c-5d5a-42a1-91b6-b9b568282bce)

**Comment by andz-bb** [comment, marked as answer]
Hey @simon-lewis

The Field Group component only expects form field components to be direct children of it. This is so that it can render the labels and input boxes properly depending on the Field Group settings. Could you move the Data Provider up a level so that the whole Field Group is within it?

**Reply by melohagan** [reply]
This is a known issue with the field group - if you set the label positioning to **above** that will also resolve this issue

**Comment by simon-lewis** [comment]
Thanks @andz-bb. Yep, tried this and it does work for one Options Picker and its associated Data Provider. Problem I have is that I need to add 4 Options Pickers to the form (replacing Relationship Pickers) so will need to have 4 Data Providers. The Field group can only be within one of them. Still - good solution for a single Options Picker. 
I think I am gonna ditch the side panel and build a separate modal for the edit of this particular entity. Appreciate the follow-up, Andrew.
